main.py

```diff
@@ -2,918 +2,30 @@
-                if title_ref_search:
-                    title = title_ref_search.group(0).strip()
-                    ref_search = self.REF_PATTERN.search(title)
-                    if ref_search:
-                        ref = ref_search.group(0)
-                        title = title.replace(ref, '', 1).lstrip(':) ')
-                    break
-        return title, ref, title_identifier
-
-    def _get_issue_url(self, comment):
-        """Check the passed comment for a GitHub issue URL."""
-        url_search = self.ISSUE_URL_PATTERN.search(comment, re.IGNORECASE)
-        url = None
-        if url_search:
-            url = url_search.group(0)
-            parsed_url = urlparse(url)
-            return url if all([parsed_url.scheme, parsed_url.netloc]) else None
-        return url
-
-    def _get_labels(self, comment):
-        """Check the passed comment for issue labels."""
-        labels_search = self.LABELS_PATTERN.search(comment, re.IGNORECASE)
-        labels = []
-        if labels_search:
-            labels = labels_search.group(0).replace(', ', ',')
-            labels = list(filter(None, labels.split(',')))
-        return labels
-
-    def _get_assignees(self, comment):
-        """Check the passed comment for issue assignees."""
-        assignees_search = self.ASSIGNEES_PATTERN.search(comment, re.IGNORECASE)
-        assignees = []
-        if assignees_search:
-            assignees = assignees_search.group(0).replace(', ', ',')
-            assignees = list(filter(None, assignees.split(',')))
-        return assignees
-
-    def _get_milestone(self, comment):
-        """Check the passed comment for a milestone."""
-        milestone_search = self.MILESTONE_PATTERN.search(comment, re.IGNORECASE)
-        milestone = None
-        if milestone_search:
-            milestone = milestone_search.group(0)
-        return milestone
-
-    # noinspection PyMethodMayBeStatic
-    def _should_ignore(self, file):
-        ignore_patterns = os.getenv('INPUT_IGNORE', None)
-        if ignore_patterns:
-            for pattern in filter(None, [pattern.strip() for pattern in ignore_patterns.split(',')]):
-                if re.match(pattern, file):
-                    return True
-        return False
+from TodoParser import *
+from LineStatus import *
+from LocalClient import *
+from GitHubClient import *
 
 
 if __name__ == "__main__":
-    # Create a basic client for communicating with GitHub, automatically initialised with environment variables.
-    client = GitHubClient()
+    # Try to create a basic client for communicating with the remote version control server, automatically initialised with environment variables.
+    try:
+        # try to build a GitHub client
+        client = GitHubClient()
+    except EnvironmentError:
+        # don't immediately give up
+        client = None
+    # if needed, fall back to using a local client for testing
+    client = client or LocalClient()
+
```

Comment:
Rather than unconditionally trying to create a GitHub client (and failing with an ugly uncaught exception when you can't), this now catches the situation where the GitHub client can't be constructed and falls back to using a (very simply) Local client.